### ISSUE TITLE
Unexpected DATAEND

### ISSUE THREAD

**Issue by OberstVonGatow** [abaplint/abaplint#2114]
I got this Error in a Class:

`Unexpected DATAEND, ML03TCOM.PROG.ABAP (structure)`

(checkname is just "... (...)" )

I don't quite get why the error appears. If I copy and paste the exact same part of the code in a temp programm, there is no error... (There are no syntax errors and)

![image](https://user-images.githubusercontent.com/6384996/131453452-68860719-e1c8-4302-b582-ace918c600f3.png)

**Comment by larshp** [comment]
the error message is not very good

something in the file, probably between DATA BEGIN OF and DATA END OF is not recognized by abaplint

try copying the contents to the playground, and delete statements until the erroneous statement is identified

**Comment by larshp** [comment]
@OberstVonGatow did the above help? I cannot debug it without the source of the full include/program/class

**Comment by OberstVonGatow** [comment]
@larshp thanks for the quick response. I think I have to fix some other issues with our configuration first. Maybe it is an issue with our docker container or my eclipse installation (at the moment i get always the wrong error title in eclipse).
After that i look into the "Unexpected DATAEND" Error. Strange is that there is no DATA BEGIN OF line in the whole class.

**Comment by OberstVonGatow** [comment]
I think i found the error. I generated Types deklaration like this:

```
CLASS zcl_test_abaplint01 DEFINITION
  PUBLIC
  FINAL
  CREATE PUBLIC .

  PUBLIC SECTION.

    TYPES:
      ty_lt_ltap_create TYPE STANDARD TABLE OF ltap_creat,
      ty_lt_ltak        TYPE STANDARD TABLE OF ltak_vb.

  PROTECTED SECTION.
  PRIVATE SECTION.

ENDCLASS.

CLASS zcl_test_abaplint01 IMPLEMENTATION.

ENDCLASS.

```
![image](https://user-images.githubusercontent.com/6384996/133594090-f06b8440-43d4-4f65-9137-b46b7f2b1f0d.png)

**Comment by OberstVonGatow** [comment]
Okay now i don't have a clue at all... Same Error empty class:

```
CLASS zcl_test_abaplint01 DEFINITION
  PUBLIC
  FINAL
  CREATE PUBLIC .

  PUBLIC SECTION.

  PROTECTED SECTION.
  PRIVATE SECTION.

ENDCLASS.



CLASS zcl_test_abaplint01 IMPLEMENTATION.

ENDCLASS.

```
![image](https://user-images.githubusercontent.com/6384996/133594737-13ff7802-054c-4f18-9365-acc915589296.png)

**Comment by OberstVonGatow** [comment]
I copied a class where i got that error and than started to delete method after method to check where the error is coming from. 
Now I end up with an empty class and still got the error ^^

**Comment by larshp** [comment]
try using https://playground.abaplint.org just copy-paste the ABAP code there, should be possible to reproduce

**Comment by OberstVonGatow** [comment]
Unfortunately it did not work in https://playground.abaplint.org. 
Copied the code and changed it to a local class also had to change a lot of dependencies.
Because the other checks are working correctly i won't invest too much time into this issue right now.
Think i will investigate if the error occurs in a different class/report.

**Comment by larshp** [comment]
closing, no progress
